syncapi/storage/postgres/peeks_table.go

```diff
@@ -0,0 +1,183 @@
+// Copyright 2020 The Matrix.org Foundation C.I.C.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package postgres
+
+import (
+	"context"
+	"database/sql"
+	"time"
+
+	"github.com/matrix-org/dendrite/internal"
+	"github.com/matrix-org/dendrite/internal/sqlutil"
+	"github.com/matrix-org/dendrite/syncapi/storage/tables"
+	"github.com/matrix-org/dendrite/syncapi/types"
+)
+
+const peeksSchema = `
+CREATE TABLE IF NOT EXISTS syncapi_peeks (
+	id BIGINT PRIMARY KEY DEFAULT nextval('syncapi_stream_id'),
```

Comment:
By having this be the primary key means we effectively don't ever clobber entries in this table. If we had it being a unique key on room/user/device then we could just toggle the id/deleted status and bound growth of this table.

Reply:
hum, this is also going to go badly with DeletePeeks, which tries to set the IDs of the deleted rows to be the same streampos, i think.

I'm missing how removing the pkey would change how the table grows though...

Reply:
> which tries to set the IDs of the deleted rows to be the same streampos

Postgres is fine because the query uses `nextval('syncapi_stream_id')` on each UPDATE. This will fail on SQLite though because as you say we do use the same stream ID.

> I'm missing how removing the pkey would change how the table grows though

It doesn't, it's just I expected to see it clobber on user/device/room instead which is what made me realise that the current form is unbounded growth.

Reply:
fixed in b1d1ccdb